src/main.rs

```diff
@@ -48,124 +80,237 @@ impl LogLevel {
-    let ss: Vec<String> = services.iter().map(|x| x.unit.clone()).collect();
-    let active_units: HashSet<String> = HashSet::from_iter(ss);
-    for service in ["sshd.service", "NetworkManager.service"] {
-        if !active_units.contains(service) {
-            log::warn!("service {} failed, see journal", service);
-            failure = true;
-        }
+
+    if script_failure {
+        bail!("health-check failed!");
     }
-    if failure {
-        for path in ["/etc/greenboot/red.d/*.sh"] {
-            for entry in glob(path)?.flatten() {
-                log::info!("running red check {}", entry.to_string_lossy());
-                let output = Command::new("bash").arg("-C").arg(entry).output()?;
-                if !output.status.success() {
-                    // combine and print stderr/stdout
-                    log::warn!("red script failed...");
-                }
+    Ok(())
+}
+
+fn run_red() -> Result<(), Error> {
+    for path in GREENBOOT_INSTALL_PATHS {
+        let red_path = format!("{path}/red.d/*.*");
+        for entry in glob(&red_path)?.flatten() {
+            log::info!("running red check {}", entry.to_string_lossy());
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
+            if !output.status.success() {
+                // combine and print stderr/stdout
+                log::warn!("red script: {} failed!", entry.to_string_lossy());
+                log::warn!("reason: {}", String::from_utf8_lossy(&output.stderr));
             }
         }
-        log::warn!("SYSTEM is UNHEALTHY. Rolling back and rebooting...");
-        Command::new("rpm-ostree").arg("rollback").status()?;
-        reboot()?;
-        return Ok(());
     }
-    for path in ["/etc/greenboot/green.d/*.sh"] {
-        for entry in glob(path)?.flatten() {
+    Ok(())
+}
+
+fn run_green() -> Result<(), Error> {
+    for path in GREENBOOT_INSTALL_PATHS {
+        let green_path = format!("{path}/green.d/*.*");
+        for entry in glob(&green_path)?.flatten() {
             log::info!("running green check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
             if !output.status.success() {
                 // combine and print stderr/stdout
-                log::warn!("green script failed...");
+                log::warn!("green script {} failed!", entry.to_string_lossy());
+                log::warn!("reason: {}", String::from_utf8_lossy(&output.stderr));
             }
         }
     }
     Ok(())
 }
 
-fn reboot() -> Result<(), Error> {
-    Command::new("systemctl").arg("reboot").spawn()?;
-    Ok(())
+fn health_check() -> Result<()> {
+    let config = GreenbootConfig::get_config();
+    log::info!("{config:?}");
+    _ = handle_motd("healthcheck is in progress");
```

Comment:
I believe it is more idiomatic to write this as

```suggestion
    handle_motd("healthcheck is in progress").ok();
```

(Note: still new to Rust and leaning on ChatGPT for help here)